src/app/thankYou/thankYou.component.js

```diff
@@ -43,6 +49,14 @@ class ThankYouController{
         if(this.purchase.paymentMeans.self.type === 'elasticpath.purchases.purchase.paymentmeans'){ //only credit card type has billing address
           this.billingAddress = this.orderService.formatAddressForTemplate(this.purchase.paymentMeans['billing-address'].address);
         }
+
+        // Display Account Benefits Modal when GUEST checkout
```

Comment:
PUBLIC user's always have a registration-state of MATCHED (They weirdly match themselves), so it will be shown.